pyVmomi/SoapAdapter.py

```diff
@@ -1094,6 +1096,24 @@ def __call__(self, path, key_file=None, cert_file=None, **kwargs):
       return retval
 
 
+## Stand-in for the HTTPSConnection class that will connect to a regular HTTP
+## proxy.
+class HTTPProxyConnection(object):
+   # @param proxyPath The path to pass to the CONNECT command.
+   def __init__(self, proxyPath):
+      self.proxyPath = proxyPath
+
+   # Connects to a HTTP proxy server and initiates a tunnel to the destination
+   # specified by proxyPath.  If successful, a new HTTPSConnection is returned.
+   #
+   # @param path The destination URL path.
+   # @param args Arguments are ignored
+   # @param kwargs Arguments for HTTPSConnection
+   def __call__(self, path, *args, **kwargs):
+      conn = http_client.HTTPSConnection(path, **kwargs)
```

Comment:
Hi @goneri , Please can you check this? here **kwargs contain the arguments which are supported by http_client.HTTPSConnection(). Instead try using SoapAdapter._HTTPSConnection()?

We see this error with your change` TypeError: __init__() got an unexpected keyword argument 'cert_reqs'`

Thanks